python/src/array.cpp

```diff
@@ -415,6 +415,50 @@ array np_array_to_mlx(py::array np_array, std::optional<Dtype> dtype) {
   }
 }
 
+std::string buffer_format(const array& a) {
+  switch (a.dtype()) {
```

Comment:
explicit string values determined via the unit test: 

```
self.assertEqual(mv_mx.format, mv_np.format, f"{mlx_dtype}{np_dtype}")
```
Not sure why `pybind11::format_descriptor` does not always agree with numpy.

Reply:
clarified and documented via comments.